homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -250,3 +268,72 @@ def device_class(self):
     def state(self):
         """Return the state of the sensor."""
         return self._sub_device.status[self._data_key]
+
+
+class XiaomiGatewayIlluminanceSensor(Entity):
+    """Representation of the XiaomiGatewayIlluminanceSensor."""
+
+    def __init__(self, gateway_device, gateway_name, gateway_device_id):
+        """Initialize the entity."""
+        self._gateway = gateway_device
+        self._name = f"{gateway_name} Illuminance"
+        self._gateway_device_id = gateway_device_id
+        self._unique_id = f"{gateway_device_id}-illuminance"
+        self._available = False
+        self._state = None
+
+    @property
+    def unique_id(self):
+        """Return an unique ID."""
+        return self._unique_id
+
+    @property
+    def device_id(self):
+        """Return the device id of the gateway."""
+        return self._gateway_device_id
+
+    @property
+    def device_info(self):
+        """Return the device info of the gateway."""
+        return {
+            "identifiers": {(DOMAIN, self._gateway_device_id)},
+        }
+
+    @property
+    def name(self):
+        """Return the name of this entity, if any."""
+        return self._name
+
+    @property
+    def available(self):
+        """Return true when state is known."""
+        return self._available
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return "lux"
+
+    @property
+    def device_class(self):
+        """Return the device class of this entity."""
+        return DEVICE_CLASS_ILLUMINANCE
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    async def async_update(self):
+        """Fetch state from the device."""
+        try:
+            self._state = await self.hass.async_add_executor_job(
+                self._gateway.get_illumination
+            )
+        except GatewayException as ex:
+            if self._available:
+                self._available = False
+                _LOGGER.error(
+                    "Got exception while fetching the gateway illuminance state: %s", ex
+                )
+        self._available = True
```

Comment:
This should go inside the try block as it gets otherwise executed all times.

Reply:
nice catch, thanks!